google/cloud/logging_v2/handlers/_helpers.py

```diff
@@ -46,59 +51,83 @@ def format_stackdriver_json(record, message):
     return json.dumps(payload)
 
 
-def get_trace_id_from_flask():
-    """Get trace_id from flask request headers.
+def get_request_data_from_flask():
+    """Get trace_id and http_request data from flask request headers.
 
     Returns:
         str: TraceID in HTTP request headers.
+        HttpRequest: data about the associated http request.
     """
     if flask is None or not flask.request:
-        return None
+        return None, None
+
+    # build http_request
+    http_request = HttpRequest(
+        request_method=flask.request.method,
+        request_url=flask.request.url,
+        request_size=flask.request.content_length,
+        user_agent=flask.request.user_agent.string,
+        remote_ip=flask.request.remote_addr,
+        referer=flask.request.referrer,
+        protocol=flask.request.environ.get(_PROTOCOL_HEADER),
+    )
 
+    # find trace id
+    trace_id = None
     header = flask.request.headers.get(_FLASK_TRACE_HEADER)
+    if header:
+        trace_id = header.split("/", 1)[0]
 
-    if header is None:
-        return None
-
-    trace_id = header.split("/", 1)[0]
-
-    return trace_id
+    return http_request, trace_id
 
 
-def get_trace_id_from_django():
-    """Get trace_id from django request headers.
+def get_request_data_from_django():
+    """Get trace_id and http_request data from django request headers.
 
     Returns:
         str: TraceID in HTTP request headers.
+        HttpRequest: data about the associated http request.
```

Comment:
Can you adjust the 'Returns' to show that this returns a Tuple? See [example](https://github.com/googleapis/google-auth-library-python/blob/0323cf390b16e8483660ac88775e8ea4e7f7702d/google/auth/_default.py#L315-L319)